drivers/soft_spi/soft_spi.c

```diff
@@ -60,50 +54,47 @@ void soft_spi_init_pins(soft_spi_t bus)
 {
     DEBUG("Soft SPI soft_spi_init_pins\n");
 
-    assert(soft_spi_bus_is_valid(bus));
+    assert(bus < SOFT_SPI_NUMOF);
 
-    /* check that miso is not mosi is not clk*/
-    assert(!gpio_is_equal(soft_spi_config[bus].mosi_pin, soft_spi_config[bus].miso_pin));
-    assert(!gpio_is_equal(soft_spi_config[bus].mosi_pin, soft_spi_config[bus].clk_pin));
-    assert(!gpio_is_equal(soft_spi_config[bus].miso_pin, soft_spi_config[bus].clk_pin));
     /* mandatory pins */
-    assert(gpio_is_valid(soft_spi_config[bus].mosi_pin) ||
-           gpio_is_valid(soft_spi_config[bus].miso_pin));
-    assert(gpio_is_valid(soft_spi_config[bus].clk_pin));
+    assert((GPIO_UNDEF != soft_spi_config[bus].mosi_pin) ||
+           (GPIO_UNDEF != soft_spi_config[bus].miso_pin) ||
+           (GPIO_UNDEF != soft_spi_config[bus].clk_pin));
+
+    /* check that miso is not mosi is not clk*/
+    assert((soft_spi_config[bus].mosi_pin != soft_spi_config[bus].miso_pin) ||
+           (soft_spi_config[bus].mosi_pin != soft_spi_config[bus].clk_pin) ||
+           (soft_spi_config[bus].miso_pin != soft_spi_config[bus].clk_pin));
 
-    /* initialize clock pin */
     gpio_init(soft_spi_config[bus].clk_pin, GPIO_OUT);
-    /* initialize optional pins */
-    if (gpio_is_valid(soft_spi_config[bus].mosi_pin)) {
-        gpio_init(soft_spi_config[bus].mosi_pin, GPIO_OUT);
-        gpio_clear(soft_spi_config[bus].mosi_pin);
-    }
-    if (gpio_is_valid(soft_spi_config[bus].miso_pin)) {
-        gpio_init(soft_spi_config[bus].mosi_pin, GPIO_IN);
-    }
+    gpio_init(soft_spi_config[bus].mosi_pin, GPIO_OUT);
+    gpio_init(soft_spi_config[bus].miso_pin, GPIO_IN);
 }
 
 int soft_spi_init_cs(soft_spi_t bus, soft_spi_cs_t cs)
 {
     DEBUG("Soft SPI init CS\n");
-    if (!soft_spi_bus_is_valid(bus)) {
+    if (bus >= SOFT_SPI_NUMOF) {
         DEBUG("Soft SPI bus not valid\n");
         return SOFT_SPI_NODEV;
     }
 
-    if (gpio_is_valid(cs) && !gpio_is_equal(cs, SOFT_SPI_CS_UNDEF)) {
-        DEBUG("Soft SPI set user CS line\n");
-        gpio_init(cs, GPIO_OUT);
-        gpio_set(cs);
+    if ((cs == GPIO_UNDEF) || (cs == SOFT_SPI_CS_UNDEF)) {
```

Comment:
Please use the `gpio_is_valid()` / `gpio_is_equal()` functions, those macros are going to go away with the GPIO API rework.